examples/cpp/core/capped_allocator/capped_allocator.cpp

```diff
@@ -0,0 +1,55 @@
+#include "capped_allocator_reflection.h"
+
+#include <bond/core/bond.h>
+#include <bond/stream/output_buffer.h>
+
+using namespace examples::capped_allocator;
+
+
+Struct PopulateStruct()
+{
+    // Use capped_allocator with unlimited size in order to populate
+    // the struct with values.
+    bond::capped_allocator<std::allocator<void> > alloc{
+        (std::numeric_limits<std::uint32_t>::max)() };
+
+    // A copy of allocator must be passed to corresponding functions.
+    Struct obj{ alloc };
+    obj.ints.push_back(1);
+    obj.strings.strings.emplace_back("test", alloc);
+    obj.strings_vector.emplace_back(alloc);
+    obj.strings_vector[0].strings.emplace_back("test2", alloc);
+
+    return obj;
+}
+
+bond::blob SerializeStruct(const Struct& obj)
+{
+    bond::OutputBuffer output;
+    bond::CompactBinaryWriter<bond::OutputBuffer> writer{ output };
+
+    Serialize(obj, writer);
+
+    return output.GetBuffer();
+}
+
+int main()
+{
+	bond::CompactBinaryReader<bond::InputBuffer> reader{
+        SerializeStruct(PopulateStruct()) };
+
+    // Limit the allowed memory allocations to 100 bytes.
+    Struct obj{ bond::capped_allocator<std::allocator<void> >{ 100 } };
+
+    try
+    {
+        Deserialize(reader, obj);
```

Comment:
We should add an assert after the Deserialize to demonstrate that Deserialize failed.

Reply:
Added.